File: Introducción a la Programación Orientada a Objetos/Ejercicio-3/Python/Coche.py
class Coche:
    def __init__(self,marca,modelo,velocidad):
        self.__marca = marca
        self.__modelo = modelo
        self.__velocidad = velocidad
    def acelerar(self):
        self.__velocidad = self.__velocidad+10
        return self.__velocidad
    def frenar(self):
        if (self.__velocidad>=5):
            self.__velocidad = self.__velocidad-5
        else:
            self.__velocidad = 0
        return self.__velocidad    
    def muestravelocidades(self):
        print(f"Marca: {self.__marca} \nModelo: {self.__modelo} \nVelocidad: {self.__velocidad} km/h")
c1 = Coche("Porsche","718 Cayman GT4 RS",249)
print("---------------------------------------------------------------------")
print("Introduccion a POO: ")
print("Ejercicio 3:")
print("---------------------------------------------------------------------")
print("Coches")
print("---------------------------------------------------------------------")
c1.muestravelocidades()
print("---------------------------------------------------------------------")
print("Velocidad Inicial: ",c1._Coche__velocidad,"km/h")
newvelocidad = c1.acelerar()
print(f"Velocidad al Acelerar: {newvelocidad} km/h")
frvelocidad = c1.frenar()
print(f"Velocidad al Frenar: {frvelocidad} km/h")
print("---------------------------------------------------------------------")
c2 = Coche("Lamborghini","Aventador LP700-4",290)
c2.muestravelocidades()
print("---------------------------------------------------------------------")
print("Velocidad Inicial: ",c2._Coche__velocidad,"km/h")
newovelocidad = c2.acelerar()
print(f"Velocidad al Acelerar: {newovelocidad} km/h")
fruvelocidad = c2.frenar()
print(f"Velocidad al Frenar: {fruvelocidad} km/h")
print("---------------------------------------------------------------------")

File: Introducción a la Programación Orientada a Objetos/Ejercicio-3/Java/src/vehiculos/Coche.java
package vehiculos;

public class Coche {
	private String marca;
	private String modelo;
	private float velocidad;
	
	public Coche(String marca,String modelo,float velocidad) {
		this.marca=marca;
		this.modelo=modelo;
		this.velocidad=velocidad;
	}
	
	public float acelerar() {
		return this.velocidad=this.velocidad+10;
	}
	
	public float frenar() {
		if(this.velocidad>=5) {
			return this.velocidad=this.velocidad-5;
		}
		else {
			return this.velocidad=0;
		}
	}
	
	public String muestracochesvelocidades() {
		return "Marca: "+this.marca+" | "+"Modelo: "+this.modelo+" | "+"Velocidad: "+this.velocidad+" km/h";
	}
	public static void main(String[] args) {
		Coche car1 = new Coche("Lamborghini","Aventador LP 700-4 Ultimae",300);
		Coche car2 = new Coche("Porsche","718 Cayman GT4 RS",249);
		System.out.println("---------------------------------------------------------------------------");
		System.out.println("Introducción a POO:");
		System.out.println("Exercise 3: ");
		System.out.println("---------------------------------------------------------------------------");
		System.out.println("Coches");
		System.out.println("---------------------------------------------------------------------------");
		System.out.println(car1.muestracochesvelocidades());
		System.out.println("---------------------------------------------------------------------------");
		System.out.println("Acelerando: "+car1.acelerar()+" km/h");
		System.out.println("Desacelerando: "+car1.frenar()+" km/h");
		System.out.println("---------------------------------------------------------------------------");
		System.out.println(car2.muestracochesvelocidades());
		System.out.println("---------------------------------------------------------------------------");
		System.out.println("Acelerando: "+car2.acelerar()+" km/h");
		System.out.println("Desacelerando: "+car2.frenar()+" km/h");
		System.out.println("---------------------------------------------------------------------------");
	}

}
